### PR TITLE
Fixes instant summons and probably a few other edge cases

### LINKED ISSUE #9937: Instant Summons can multiply embedded items
Server revision compiled on: 2015-06-13
cf46c8a8a5089d94acbba928e094aff06cff73cc 

Mark an object that can be embedded for recall (GAR glasses work I think, spears work for sure)
Throw the object at someone and hope it gets stuck in a limb
Recall it
You now have the object in your hand and you can throw it again to get it stuck again, while at the same time it's still stuck in the person
Eventually the object falls out of the limb and you have the one in your hand and the one that fell out + any other that you might have embedded from throwing
Only tested with spears but I am pretty sure it works with all objects that can get stuck in limbs

### LINKED ISSUE #22616: Instant summon spell lets you use marked items as any type of mob.
Mark an item and change/get changed into a different mob like an alien hunter or an ash drake, summon the item and bam, you can use that item.

### LINKED ISSUE #22837: Instant summons allows you to move anchored machinery/objects.
If you place an instant summoned object into a suit storage unit, microwave, destructive analyzer, or any anchored machinery that holds items, it teleports the anchored object to you. This also allows instant summons to steal the actual paradox bag itself and not the gimmick satchels.

### PR DESCRIPTION
**Pull request by ChangelingRain**
Fixes #22616
Fixes #22837

:cl: Joan
bugfix: Instant Summons is no longer greedy with containers.
/:cl:

Hey, look at this proc forceMove calls! maybe we should _USE FUCKING FORCEMOVE SO THAT GETS CALLED!_

### CODE REVIEW

**Comment by Kromgar** [comment]
So no more summoning containers?

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -176,6 +176,11 @@ Class Procs:
 			L.update_canmove()
 	occupant = null
 
+/obj/machinery/Exited(atom/movable/M, atom/newloc)
```
Comment (RemieRichards) [inline code/game/machinery/machinery.dm]: don't label a movable as just ```M```, the standard is ```AM```

File: code/game/machinery/suit_storage_unit.dm [hunk 2]
```diff
@@ -168,6 +168,18 @@
 	storage = null
 	occupant = null
 
+/obj/machinery/suit_storage_unit/Exited(atom/movable/M, atom/newloc)
+	if(M == helmet)
+		helmet = null
+	if(M == suit)
```
Comment (RemieRichards) [inline code/game/machinery/suit_storage_unit.dm]: ```else if```s, I would consider it a bug if an item was both the helmet suit and mask.

**Comment by Shadowlight213** [comment]
Let's have exited() call unequip()! Which calls forcemove(). Which calls exited(). which calls unequip().....

**Comment by Cyberboss** [comment]
@lzimann Reopen the two issues in the OP

**Pull request merged**